esquery.js

```diff
@@ -224,19 +237,55 @@ function matches(node, selector, ancestry) {
     throw new Error(`Unknown selector type: ${selector.type}`);
 }
 
+/**
+ * Get visitor keys of a given node.
+ * @param {external:AST} node node The AST node to get keys.
+ * @param {ESQueryOptions|undefined} options
+ * @returns {string[]} Visitor keys of the node.
+ */
+function getVisitorKeys(node, options) {
+    const nodeType = node.type;
+    let candidates;
+    if (options && options.visitorKeys && options.visitorKeys[nodeType]) {
+        candidates = options.visitorKeys[nodeType];
+    } else {
+        candidates = estraverse.VisitorKeys[nodeType];
+    }
+    if(!candidates) {
+        if (options && typeof options.fallback === 'function') {
+            candidates = options.fallback(node);
+        } else {
+            // 'iteration' fallback
+            candidates = Object.keys(node);
+        }
+    }
+    return candidates;
```

Comment:
I would structure this function as early returns instead of a bunch of assignments.

```suggestion
    if (options && options.visitorKeys && options.visitorKeys[nodeType]) {
        return options.visitorKeys[nodeType];
    }
    if (estraverse.VisitorKeys[nodeType]) {
        return estraverse.VisitorKeys[nodeType];
    }
    if (options && typeof options.fallback === 'function') {
        return options.fallback(node);
    }
    // 'iteration' fallback
    return Object.keys(node);
```